### PR TITLE
fix(connect): configure plugin MCP servers like connectors and retire them with the plugin

### PR DESCRIPTION
**Pull request by reachjalil**
## Summary
- Adding an MCP server to a plugin now asks the same connector setup as the **Connectors** page — authentication (OAuth / API key / None), whose account the AI uses, and an optional pre-registered OAuth app — and `POST /v1/plugins` accepts that setup on an `mcp` component so the server is bound to a plugin-owned connection the moment the plugin exists.
- Archiving a plugin now retires the connection it created: `GET /v1/mcp-connections?scope=manageable` omits a plugin-owned connection while every plugin bound to it is archived or deleted, and lists it again on restore.
- The segmented authentication / account controls and their copy move into one shared module (`mcp-connection-form-controls.tsx`) so the Connectors form and the plugin editor cannot drift.
- **Desktop** (follow-through on Warden 2U2-S6N): the Library's "Add organization MCP" dialog and plugin builder ask the same questions for owners/admins and forward them as the component's `connection`, so the app is not the one surface still producing unconfigured servers. Members keep the declaration-only flow Den allows them.

## Why
- The plugin editor only captured a URL. Every plugin-declared MCP server therefore landed in "Needs connection" until an admin found the collection page and configured it in a second, differently shaped dialog.
- `setPluginLifecycle` revoked sourced grants on archive but left the plugin-created `external_mcp_connection` row in the admin list. Because provenance is computed from *active* bindings only, the archived plugin's connection reappeared as an ordinary unmanaged connector with empty `requiredBy` / `identityManagedBy` — editable and confusing.

## Issue
- N/A (reported directly)

## Scope
- `den-api`
  - `pluginMcpConnectionSetupSchema` (shared by the new inline `components[].connection` and the existing configure route); `connection` is only accepted on `mcp` components.
  - `createPluginBundle` configures each declared server through the existing `configureMarketplacePluginMcpRequirement` after grants and collection are in place. A failed setup rolls back like a failed GitHub import (drop bindings and connections this attempt created, archive the plugin) so the name is free for the corrected retry. Owners/admins only; the agent-secret guard from the configure route applies.
  - `listRetiredPluginOwnedExternalMcpConnectionIds` + filter in the `scope=manageable` list. Connections an admin created directly, or that another active plugin still depends on, are never retired. Rows are hidden, not deleted, so restore works.
- `den-web`
  - Plugin editor: connector fields under the MCP URL (`McpConnectionFields`), request mapping in `plugin-mcp-connection-draft.ts`.
  - Connectors screen: uses the shared controls; no behaviour change.
- `apps/app`
  - `library.ts`: `LibraryMcpConnectionForm` → `libraryMcpConnectionRequest()`; `connection` on `LibraryPluginComponentDraft` / `CreateLibraryItemInput` / `DenLibraryPluginCreateRequest`; `createOrgPlugin()` types the field.
  - `add-library-item-modal.tsx`: `McpConnectionFields` under the server URL (single MCP and plugin components), shown only when `canConfigureMcpConnections` (admin role from the cloud session). New `extensions.add_mcp_*` strings in `en.ts` (other locales fall back, as recent string additions do).
- Tests: `evals/specs/plugin-mcp-connector-parity.test.ts` (real Den journey incl. posting the Desktop-built body), `ee/apps/den-api/test/plugin-mcp-connector-parity.test.ts` (seeded edge cases), `ee/apps/den-web/tests/plugin-mcp-connection-draft.test.ts`, `apps/app/tests/library-destination.test.ts` (forwarding + mapping).

## Out of scope
- Editing an existing plugin's MCP connector setup from the plugin detail page (the collection page "Configure" flow still covers that).
- Translating the new Desktop strings beyond `en` (CI's i18n audit does not require it; locales fall back).
- Hiding retired plugin-owned connections from member-facing views: sourced access is already revoked on archive; a direct grant an admin added by hand on a plugin-owned connection is left as is.
- Regenerating `packages/docs/openapi.json`: the committed snapshot is already stale against `dev` (unrelated SSO/admin drift), so a regeneration would pull ~500 unrelated lines into this PR.

### Ran
- `OPENWORK_EVAL_DAYTONA=1 OPENWORK_EVAL_REF=<head> pnpm evals:pr specs/plugin-mcp-connector-parity.test.ts`
- `pnpm evals:pr specs/plugin-mcp-connector-parity.test.ts` (local cold-boot `server()`)
- `pnpm --filter @openwork-ee/den-api exec bun test --conditions development test/plugin-mcp-connector-parity.test.ts` (scratch MySQL database created with `db:push`)
- `bun test --conditions development test/marketplace-cloud-readiness.test.ts`, `test/mcp-connections-required-by.test.ts`, `test/mcp-connections-edit.test.ts`, `test/plugin-system-member-create.test.ts`, `test/github-plugin-import-schema.test.ts` (den-api, one file per process)
- `pnpm --filter @openwork-ee/den-api test` (default CI lane)
- `bun test --conditions development tests/plugin-mcp-connection-draft.test.ts tests/plugin-import-flow.test.ts tests/marketplace-mcp-readiness.test.ts tests/mcp-connection-editing.test.ts tests/mcp-connection-smart-add.test.ts tests/mcp-credential-autofill.test.ts` (den-web)
- `pnpm --filter @openwork/app test`, `pnpm --filter @openwork/app typecheck`, `node scripts/i18n-audit.mjs --ci`
- `tsc --noEmit` for den-api and den-web

### Result
- pass/fail: **Passed** on head `f4e7d7af0` (see the test-evidence comment for the exact-head Daytona run).
  - Daytona lane: `plugin-mcp-connector-parity` 1 passed / 0 failed / 0 skipped, exit 0 (sandbox provisioned from this ref, deleted after). Claims: inline setup configures the connector; declaration-only creates nothing; the Desktop-built body is accepted and configured with its OAuth client; archive hides; restore lists again.
  - Local lane, same spec: 1 passed, exit 0.
  - den-api `plugin-mcp-connector-parity.test.ts`: 5 pass / 0 fail. Negative control: disabling the manageable-list filter makes the archive test fail (`Expected: false, Received: true`).
  - den-api neighbours: 33/0, 2/0, 20/0, 8/0, 1/0; default lane 21/0. den-web: 41/0. app `library-destination` 20/0; app typecheck and i18n audit clean.
- Pre-existing, unrelated (identical on pristine `dev` at `8b786a4a8`): `apps/app` lane shows 5 failures locally (capability inventory ×2, cloud boot takeover, provider-sync logout, extensions sidebar) — 1151/5 on `dev`, 1153/5 here; CI's `openwork-tests` is green. `ee/apps/den-api/test/plugin-system-create-bundle.test.ts` reports 1 fail from an unhandled better-auth init error against its stubbed DB; its schema assertions were re-checked by hand and hold with the new `connection` field.

## CI status
- pass: pending at open
- code-related failures: none known
- external/env/auth blockers: none

## Manual verification
1. Dashboard → Plugins → Create a plugin → add an **MCP server**: the URL field is followed by Authentication, API key / OAuth app, and "Whose account does the AI use?" with the same options and copy as Connectors → Add. In the Desktop app, Library → Add → **Add organization MCP** (or a plugin's MCP component) shows the same questions when you are an org owner/admin, and none of them as a member.
2. Create it. Connectors lists `<Plugin> / <server>` immediately with the chosen auth and credential mode, managed by the plugin; the collection page shows no "Needs connection" for it.
3. Archive the plugin: the connector leaves the Connectors list. Restore it: the connector is back with the plugin as identity manager. A connector added directly under Connectors is unaffected throughout.

## Evidence
- Testkit evidence for the current head is published on this PR (test-evidence comment).

## Risk
- Low. The inline `connection` field is optional; requests without it behave exactly as before, and the Desktop sends it only for owners/admins (an older Den ignores unknown component fields the same way the schema did before). The manageable-list filter only removes rows whose binding carries `connection_owned_by_plugin = true` and whose bound plugins are all non-active; member-facing lists and capability search are untouched.

## Rollback
- Revert the commit. No schema migration; no data is deleted by the new list filter.

### CODE REVIEW

**Comment by reachjalil** [comment]
## Test evidence — a-plugin-s-mcp-server-is-configured-like-a-connector-and-leaves-the-connectors-l — 0/0 screenshots passed · 3 assertions

✅ **0/0 screenshots passed · 3 assertions** · 3 expectations passed · 0 failed · 0 pending

> screenshots not uploaded (no BLOB_READ_WRITE_TOKEN)

### ℹ️ ASSERTION — 1. Adding an MCP server to a plugin configures its connection with the connector setup

POST /v1/plugins with an mcp component carrying { authType: oauth, credentialMode: shared } created &quot;Configured CRM mtkq867b5ln / crm&quot; as an OAuth, one-org-account connection owned by the plugin; scope=manageable listed it under that plugin. A declaration without connection setup created no connection.

- ✅ **PASS** Adding an MCP server to a plugin configures its connection with the connector setup — POST /v1/plugins with an mcp component carrying { authType: oauth, credentialMode: shared } created &quot;Configured CRM mtkq867b5ln / crm&quot; as an OAuth, one-org-account connection owned by the plugin; scope=manageable listed it under that plugin. A declaration without connection setup created no connection.

### ℹ️ ASSERTION — 2. The Desktop Library posts the same connector setup Den&#39;s plugin editor does

denLibraryPluginCreateRequest(&quot;mcp&quot;, …) — the body the app&#39;s &quot;Add organization MCP&quot; dialog sends — carried { authType: oauth, credentialMode: shared, oauthClient } for &quot;Desktop Linear mtkq867b5ln&quot;; Den accepted it and listed the connection as OAuth, one org account, with the pre-registered client id configured and the plugin as identity manager.

- ✅ **PASS** The Desktop Library posts the same connector setup Den&#39;s plugin editor does — denLibraryPluginCreateRequest(&quot;mcp&quot;, …) — the body the app&#39;s &quot;Add organization MCP&quot; dialog sends — carried { authType: oauth, credentialMode: shared, oauthClient } for &quot;Desktop Linear mtkq867b5ln&quot;; Den accepted it and listed the connection as OAuth, one org account, with the pre-registered client id configured and the plugin as identity manager.

### ℹ️ ASSERTION — 3. A plugin-owned connector follows the plugin&#39;s archive and restore

After POST /v1/plugins/:id/archive the manageable Connectors list no longer contained the plugin-owned connection; after POST /v1/plugins/:id/restore it was listed again with the plugin as its identity manager.

- ✅ **PASS** A plugin-owned connector follows the plugin&#39;s archive and restore — After POST /v1/plugins/:id/archive the manageable Connectors list no longer contained the plugin-owned connection; after POST /v1/plugins/:id/restore it was listed again with the plugin as its identity manager.

---
_Test run created 2026-09-02T23:27:52.656Z · Source: `evals/results/test-runs/2026-09-02T23-27-52-656Z-a-plugin-s-mcp-server-is-configured-like-a-connector-and-leaves-the-connectors-l/test-run.json` · Repro: `pnpm --dir evals artifacts:publish -- --pr 4338 --test-run 2026-09-02T23-27-52-656Z-a-plugin-s-mcp-server-is-configured-like-a-connector-and-leaves-the-connectors-l`_

File: ee/apps/den-api/src/routes/org/plugin-system/routes.ts [hunk 1]
```diff
@@ -731,10 +731,18 @@ export function registerPluginArchRoutes<T extends { Variables: OrgRouteVariable
...
+        const sessionId = c.get("session")?.id
+        if (body.components?.some((component) => component.connection && isAgentPluginMcpSecretSetup({
+          apiKey: component.connection.apiKey,
+          oauthClient: component.connection.oauthClient,
+          sessionId,
+        }))) {
+          return c.json({ error: "invalid_request", message: "Plugin MCP credentials cannot be set from the agent. Add them in the OpenWork Cloud dashboard under Connections." }, 400)
+        }
         return c.json({
           ok: true,
```
[the remark this answers is not in the document]
Comment (reachjalil) [reply ee/apps/den-api/src/routes/org/plugin-system/routes.ts:743]: Addressed in f4e7d7af0. The Desktop Library now models and forwards the setup: `LibraryPluginComponentDraft.connection` / `CreateLibraryItemInput.connection` map through `libraryMcpConnectionRequest()` into the component's `connection`, and `createOrgPlugin()` types it. The "Add organization MCP" dialog and the plugin builder show the same Connectors questions (authentication, API key, optional pre-registered OAuth app, whose account the AI uses) — for owners/admins only, since Den refuses inline setup from members, who keep today's declaration-only flow. `plugin-mcp-connector-parity.test.ts` now posts the body `denLibraryPluginCreateRequest("mcp", …)` builds to a real Den and checks the connector comes back configured with the OAuth client.
